analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd

```diff
@@ -0,0 +1,318 @@
+
+This subtype harbors either a MYB-QKI fusion or other MYB or MYBL1 fusion
+```{r}
+myb_biospecimen <- lgat_fus %>%
+  filter(MYB_fus=="Yes") %>%
+  mutate(molecular_subtype = "MYB/MYBL1") %>%
+  select("Kids_First_Biospecimen_ID","molecular_subtype")
+```
+
+### LGG, CDKN2A/B
+
+This subtype harbors focal CDKN2A and/or CDKN2B deletion
+This is a secondary co-occurring alteration with prognostic significance.
+```{r}
+
+cdkn_biospecimen <- lgat_cnv %>%
+  filter(CDKN2A_DEL=="Yes"| CDKN2B_DEL == "Yes") %>%
+  mutate(molecular_subtype = "CDKN2A/B") %>%
+  select("Kids_First_Biospecimen_ID","molecular_subtype") 
+```
+
+### Compile
+```{r}
+
+all_dna_subtype <- bind_rows(nf1_biospecimen,
+                         brafV600_biospecimen,
+                         otherMAPK_snv_biospecimen,
+                         rtk_snv_biospecimen,
+                         fgfr_cnv_biospecimen,
+                         fgfr_snv_biospecimen,
+                         idh_biospecimen,
+                         h3_biospecimen,
+                         cdkn_biospecimen
+) %>%
+  rename(Kids_First_Biospecimen_ID_DNA=Kids_First_Biospecimen_ID) %>%
+  # add Kids_First_Biospecimen_ID_RNA to dna subtypes dataframe
+  left_join(all_lgat_matched,by="Kids_First_Biospecimen_ID_DNA") 
+
+all_rna_subtype <- bind_rows(kiaabraf_biospecimen,
+                             otherMAPK_fus_biospecimen,
+                             rtk_fus_biospecimen,
+                             fgfr_fus_biospecimen,
+                             myb_biospecimen) %>%
+  rename(Kids_First_Biospecimen_ID_RNA=Kids_First_Biospecimen_ID) %>%
+  # add Kids_First_Biospecimen_ID_DNA to rna subtypes dataframe
+  left_join(all_lgat_matched,by="Kids_First_Biospecimen_ID_RNA")
+
+  
+all_subtype <- bind_rows(all_rna_subtype,
+                         all_dna_subtype) %>%
+  full_join(all_lgat_matched, by=c("Kids_First_Biospecimen_ID_RNA","Kids_First_Biospecimen_ID_DNA",
+                                   "sample_id",
+                                   "Kids_First_Participant_ID")) %>%
+  # lets group because there are some instances where LGAT DNA/RNA have multiple subtypes
+  group_by(sample_id,Kids_First_Participant_ID,Kids_First_Biospecimen_ID_RNA,Kids_First_Biospecimen_ID_DNA) %>%
+  # merging them here with summarise
+  summarise(molecular_subtype= toString(molecular_subtype)) %>%
+  # formatting molecular_subtype columns
+  dplyr::mutate(molecular_subtype = case_when(
+    # molecular_subtype is NA and we didn't find any canonical
+    # LGG variant in the DNA and no RNA abberations in RNA sample
+    # subtype should be WT
+    (molecular_subtype=="NA" & !is.na(Kids_First_Biospecimen_ID_DNA) &
+       !is.na(Kids_First_Biospecimen_ID_RNA)) ~ "LGG, wildtype",
+    # molecular_subtype is NA and we do not have a DNA sample
+    # OR don't have RNA to check for LGG variant or fusion
+    # so status will be To be classified
+    molecular_subtype=="NA" & (!is.na(Kids_First_Biospecimen_ID_DNA) |
```

Comment:
Based on the comment I would expect the check to be `is.na()`, rather than `!is.na()`
```suggestion
    molecular_subtype=="NA" & (is.na(Kids_First_Biospecimen_ID_DNA) |
```

When those columns have `NA` values that means there is no sample, is that correct?

Reply:
that's right, let me update that.